src/semeio/forward_models/design2params/design2params.py

```diff
@@ -98,13 +98,24 @@ def _complete_parameters_file(
     )
 
     # merge realization from design-matrix with parameters and defaults to one dataframe
+    # For pandas>=2.2 we need to make sure that merge preserve the original order
+    parameters.reset_index(names="order_param", inplace=True)
+    realization_values.reset_index(names="order_real", inplace=True)
+    default_sheet.reset_index(names="order_default", inplace=True)
     merged = pd.merge(
         parameters,
-        pd.merge(realization_values, default_sheet, on="keys", how="outer"),
+        pd.merge(
+            realization_values,
+            default_sheet,
+            on="keys",
+            how="outer",
+            copy=False,
+        ),
         on="keys",
         how="outer",
+        copy=False,
     )
-
+    merged.sort_values(["order_param", "order_real", "order_default"], inplace=True)
```

Comment:
If I change the order of "order_param" and "order_real" on this line, I will not fail the tests. Can you add sufficient tests to assert that what this line specifies cannot be changed?

Reply:
It might not be important, but it would be nice to have it specified by a test so we would know if it changes later. The process of writing the test is probably the way to figure out if it is important or not :)

Reply:
Nice. It only applies to logging then I reckon? (which one could argue is not as important..)